checker/src/abstract_value.rs

```diff
@@ -2982,7 +3012,19 @@ impl AbstractValueTrait for Rc<AbstractValue> {
                 if let Some(val) = environment.value_at(&path) {
                     val.clone()
                 } else {
-                    let refined_path = path.refine_paths(environment);
+                    let refined_path = path.refine_paths(pre_environment, environment);
+                    if let PathEnum::QualifiedPath {
+                        qualifier,
+                        selector,
+                        ..
+                    } = &refined_path.value
+                    {
+                        if *selector.as_ref() == PathSelector::CopyOfOriginal {
+                            if let Some(val) = pre_environment.value_at(qualifier) {
+                                return val.clone();
+                            }
+                        }
```

Comment:
Should this routine be applied to UnknownModelField and UnknownTagField, because these two cases also contain an environment lookup?